pkg/products/amqonline/reconciler.go

```diff
@@ -369,17 +369,27 @@ func (r *Reconciler) reconcilePrometheusRule(ctx context.Context, installation *
 		},
 	}
 
-	rules := []monitoringv1.Rule{}
-	rules = append(rules, monitoringv1.Rule{
-		Alert: fmt.Sprintf("AMQ-SLI-1.1: AMQ Online console is not available in namespace %s", r.Config.GetNamespace()),
-		Annotations: map[string]string{
-			"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
-			"message": "AMQ Online Console is not available",
+	rules := []monitoringv1.Rule{
+		{
+			Alert: fmt.Sprintf("AMQ-SLI-1.1: AMQ Online console is not available in namespace %s", r.Config.GetNamespace()),
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": "AMQ Online Console is not available",
+			},
+			Expr:   intstr.FromString(fmt.Sprintf("absent(kube_endpoint_address_available{endpoint=\"console\",namespace=\"%s\"}==1)", r.Config.GetNamespace())),
+			For:    "60s",
+			Labels: map[string]string{"severity": "critical"},
+		}, {
+			Alert: fmt.Sprintf("AMQ-SLI-1.3: Keycloak is not available in namespace %s", r.Config.GetNamespace()),
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": "Keycloak is not available, addresses requiring user authentication will not receive messages",
+			},
+			Expr:   intstr.FromString(fmt.Sprintf("absent(kube_endpoint_address_available{endpoint=\"standard-authservice\",namespace=\"%s\"}==2)", r.Config.GetNamespace())),
```

Comment:
Here the conditional has to be set as 2, as this is the default value being returned by the query, this is because there are indeed two containers as seen below on 
![image](https://user-images.githubusercontent.com/35736504/74176372-73e85000-4c2f-11ea-96df-8252e6e7524d.png)
 @grdryn  Should I maybe implement a conditional to 1 or 2 containers? I'm not sure whether the init container gets deleted or if anything happens to it after it's finished, that may cause this alert to misfire?

Reply:
@HubertStefanski An "Init Container" is just run to completion at the start, before the regular container(s) run(s). It can be used for things like ensuring pre-requisite services are available.

I think the value here is 2 instead because there are two ports exposed from that one regular container:

![image](https://user-images.githubusercontent.com/442386/74191708-04348e00-4c4c-11ea-857c-ed97a7d27413.png)

Reply:
@grdryn Guess that makes more sense, Would also explain why the query picks up on 2 endpoints, So it's completely fine to leave it at two?

Reply:
Yes, but maybe declare that as a named constant, so that we know why it's 2. Something like `KeycloakServicePortCount`.

https://en.wikipedia.org/wiki/Magic_number_%28programming%29#Unnamed_numerical_constants